javascript.js

```diff
@@ -11,4 +11,288 @@ document.getElementById('open-menu').onclick = openNav;
+  popupdescription: `Lorem Ipsum is simply dummy text of the printing and typesetting 
+  industry. Lorem Ipsum has been the industry's standard dummy text ever since the 
+  1500s, when an unknown printer took a galley of type and scrambled it 1960s with the
+   releaLorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem
+    Ipsum has been the industry's standard dummy text ever since the 1500s, when an 
+    unknown printer took a galley of type and scrambled it 1960s with the releax map 
+    lapora verita.`,
+  tecnologies: ['CSS', 'HTML', 'Bootstrap', 'Ruby'],
+  live: '#',
+  source: '#',
+};
+
+const projectsArray = [
+  project0,
+  project1,
+  project2,
+  project3,
+  project4,
+  project5,
+  recentWorks,
+];
+
+function popup(num) {
+  document.getElementById('popup').innerHTML = `
+  <div class="darkpage"></div>
+  <div class="popup-container" >
+    <button id="close-popup" class="close-pu" type="button">&times;</button>
+    <h3 class="works-title-2 title-pu">${projectsArray[num].title}</h3>
+    <ul class="ul-p ul-pu">  
+      <li class="li-p li-pu">${projectsArray[num].tecnologies[0]}</li>  
+      <li class="li-p li-pu">${projectsArray[num].tecnologies[1]}</li>  
+      <li class="li-p li-pu">${projectsArray[num].tecnologies[2]}</li>
+    </ul>
+    <div class="flex-popup">
+    <img class="popup-img" src=${projectsArray[num].imagepopup}
+    alt="popup image" width="311" height="220"></img>
+    <div class="popup-right">
+    <p class="p-Project p-popup">${projectsArray[num].popupdescription}</p>
+    <div class="popup-btn-container">
+    <button type="button" class="popup-btn" href=${projectsArray[num].live}>
+    See Live <i class="fas fa-external-link-alt darkicon"></i></button>
+    <button type="button" class="popup-btn btn-ml" href=${projectsArray[num].source}>
+    See Source <i class="fab fa-github darkicon" title="github icon"></i></button>
+    </div>
+    </div>
+    </div>
+  </div>`;
+
+  document.querySelector('html').style.overflow = 'hidden';
+  const startpopup = document.getElementById('popup');
+  function erase() {
+    while (startpopup.firstChild) {
+      startpopup.removeChild(startpopup.lastChild);
+    }
+    document.querySelector('html').style.overflow = 'scroll';
+  }
+
+  document.getElementById('close-popup').addEventListener('click', () => {
+    erase();
+  });
+}
+document.getElementById('recentWork').innerHTML = `
+<img class="img-works" width="327" height="237" alt="works image" src=${projectsArray[6].image}>
+<div class="works-info">
+<h3 class="works-title-2">${projectsArray[6].title}</h3>
+<p class="p-work">${projectsArray[6].paragraph}</p>
+<ul>
+  <li class="li2">${projectsArray[6].tecnologies[0]}</li>
+  <li class="li2">${projectsArray[6].tecnologies[1]}</li>
+  <li class="li2">${projectsArray[6].tecnologies[2]}</li>
+  <li class="li2">${projectsArray[6].tecnologies[3]}</li>
+</ul>
```

Comment:
- [ ] [OPTIONAL] Hi, in a situation like these you can avoid hardcoding the index values by using the `map` function to iterate through however many entries are in the array. that way it will be more scalable and less error-prone. In a situation where you have to projects having a different number of project technologies, your code might crash. looping through the `technologies` will ensure that this code could always run regardless of how many technologies they are